.github/workflows/release.yml

```diff
@@ -43,6 +53,10 @@ jobs:
               --detach-sign \
               --output pie.phar.asc \
               pie.phar
+      - name: Generate build provenance attestation
+        uses: actions/attest-build-provenance@v1
```

Comment:
If I'm understanding correctly, this should entirely negate the need to "manually" sign the phar, right? Or is there something I'm missing? I'm wondering if we can rip out the GPG signing stuff, and update the documentation for the verification to use the `gh attestation verify ...` command.

Reply:
Yes, this is my understanding. The only possible drawback would be that PGP is much more easily available compared to GitHub CLI. Debian for example bundles GitHub CLI only since the newest stable version and only in version 2.23, which does not yet include the `gh attestation` sub-command.

I'm also just noticing that the attestation functionality also has support for [attesting the SBOM](https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds#generating-an-attestation-for-a-software-bill-of-materials-sbom), which might be an interesting follow-up.

Reply:
I think needing `gh` tool is an acceptable tradeoff. The attestations I believe are accessible via an API call anyway, which `gh` wraps, so in theory another tool could be used to fetch that.

I'll do this in a separate PR, as I want to consolidate the release pipelines too.

Reply:
#132